tests/unit/utils/Utils.cpp

```diff
@@ -68,3 +70,18 @@ char * mktemp_cst(char * __template)
 
     return __template;
 }
+
+char* generate_file_name(char* file_name)
```

Comment:
mmm, maybe I misunderstand something in here. If I pass "serialization-asciiCDynamicArray.XXXXXX" directly to `generate_file_name`, it will generate warning like:

>
/Users/mikeling/shogun/tests/unit/lib/DynamicArray_unittest.cc:266:42: warning: ISO C++11 does
      not allow conversion from string literal to 'char *' [-Wwritable-strings]
        char* filename = generate_temp_filename("serialization-asciiCDynamicArray.XXXXXX");
>